dask/dataframe/core.py

```diff
@@ -1432,23 +1435,39 @@ def to_dask_array(self, lengths=None, meta=None):
               on the first axis. These values are *not* validated for
               correctness, beyond ensuring that the number of items
               matches the number of partitions.
+        dtype : str or numpy.dtype, optional
+            The dtype to pass to :meth:`numpy.asarray` when called on
+            each partition.
+        na_value : Any, optional
+            The value to use for missing values. The default value depends
+            on `dtype` and the dtypes of the DataFrame columns.
+
+            Requires pandas >= 1.1.0.
         meta : object, optional
             An optional `meta` parameter can be passed for dask to override the
-            default metadata on the underlying dask array.
+            default metadata on the returned dask array.
+            Should not be used outside of advanced cases; setting `dtype`
+            is preferred.
 
         Returns
         -------
         """
+        if not PANDAS_GT_110 and na_value is not pd_no_default:
+            raise NotImplementedError(
+                "na_value is not a valid argument for to_dask_array"
+                f"if pandas < 1.1.0. Pandas version is {pd.__version__}"
+            )
+
         if lengths is True:
             lengths = tuple(self.map_partitions(len, enforce_metadata=False).compute())
 
-        arr = self.values
+        arr = self.map_partitions(methods.to_numpy, _dtype=dtype, na_value=na_value)
```

Comment:
cc @quasiben will this cause problems with a cuFD-backed Dask DataFrame? I'm actually not sure what happens today with `to_dask_array` in that case

Reply:
Yes this will. I'm not sure what we should do on the cuDF front here because we could either have `to_numpy` return a CuPy array, or have it do an explicit device to host copy, neither of which is clear and good behavior.

Reply:
Today, `.values` will return a CuPy array which can live behind a dask array happily.

Reply:
If this is absolutely necessary, we could have `method.to_numpy` currently return `df.values` if `"cudf" in str(type(df))`, for example, but of course this will look a strange.

Reply:
Thanks for the input @kkraus14 @pentschev. We were worried about that.

I opened https://github.com/dask/dask/pull/7399 to implement "approach 2" as given in this PR's description. That change would allow us to use `.values` here instead of `to_numpy`, which is sounding like the better option to me.

@jrbourbeau we were worried that pandas ExtensionArrays might not currently behave well as dask Array chunks (particularly since they're 1D only), hence the `to_numpy` cast. Since it's seeming like `.values` is the more correct approach overall, should I spend a little time seeing how ExtensionArrays actually behave as chunks, or should we go with @pentschev's `"cudf" in str(type(df))` trick for now, merge this, and revisit it later?

Reply:
@gjoseph92, @jsignell, @jrbourbeau, and I met today. We discussed this. A few other ideas we came up with are

* Using `.array` ( https://github.com/dask/dask/issues/5001 ). Though this is `Series` only AFAICT and I don't think we implement this in cuDF to (feel free to correct me though)
* Calling `.fillna` beforehand and continuing to use `.values`. Possibly concerns about performance. Though this should work with Pandas and cuDF today. Though I think this doesn't work quite correctly for some Pandas ExtensionArrays (feel free to correct me if I got that wrong Gabe)

Feel free to add anything else or correct anything I got wrong above

Reply:
Should add I think a 3rd option (in addition to the other ideas above and ones discussed) might be Dask provides a dispatch function that Pandas and cuDF implementations can be registered with